relay-monitors/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! Monitors protocol and processing for Sentry.
+//!
+//! [Monitors] allow you to monitor the uptime and performance of any scheduled, recurring job in
+//! Sentry. Once implemented, it'll allow you to get alerts and metrics to help you solve errors,
+//! detect timeouts, and prevent disruptions to your service.
+//!
+//! # API
+//!
+//! The public API documentation is available on [Sentry Docs](https://docs.sentry.io/api/crons/).
+//!
+//! [monitors]: https://docs.sentry.io/product/crons/
+
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png",
+    html_favicon_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png"
+)]
+#![warn(missing_docs)]
+
+use relay_common::Uuid;
+use serde::{Deserialize, Serialize};
+
+/// Error returned from [`process_check_in`].
+#[derive(Debug, thiserror::Error)]
+pub enum ProcessCheckInError {
+    /// Failed to deserialize the payload.
+    #[error("failed to deserialize check in")]
+    Json(#[from] serde_json::Error),
+}
+
+///
+#[derive(Clone, Copy, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+enum CheckInStatus {
+    /// Check-in had no issues during execution.
+    Ok,
+    /// Check-in failed or otherwise had some issues.
+    Error,
+    /// Check-in is expectred to complete.
+    InProgress,
+    /// Monitor did not check in on time.
+    Missed,
+    /// No status was passed.
```

Comment:
```suggestion
    /// No status was passed.
    ///
    /// When we enable check-in deserialization in external relays, this should become `Unknown(String)` for forward compatibility.
```

Reply:
Not necessarily, that depends on the kind of normalization we're doing in external Relays. It may be safer to keep it as "Unknown" rather than applying the wrong normalization (same to the metrics extraction version).